src/core/worker.js

```diff
@@ -447,6 +447,55 @@ class WorkerMessageHandler {
         .then(page => pdfManager.ensure(page, "jsActions"));
     });
 
+    handler.on("GetEditableAnnotations", async function ({ pageIndexToSkip }) {
+      const [numPages, annotationGlobals] = await Promise.all([
+        pdfManager.ensureDoc("numPages"),
+        pdfManager.ensureDoc("annotationGlobals"),
+      ]);
+
+      if (!annotationGlobals) {
+        return null;
+      }
+      const types = new Set(["FreeText", "Stamp", "Ink", "Highlight"]);
```

Comment:
Is there a way to avoid hardcoding this list here? Can we use the annotation constants for example? I'm a bit concerned it's easy to forget to update this if we add support for more annotation types.

Reply:
You're right about having to maintain this list... So I removed the set of types and made it a parameter. The function will be used in tools.js when switching to editing mode.